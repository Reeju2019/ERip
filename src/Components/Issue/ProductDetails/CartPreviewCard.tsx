interface ICartPreviewCardProps {
  item: {
    model?: string
    serviceImage: string
    serviceName: string
    price: number
    off: number
    repairTime: string
    warranty: string
    repairType: string
  }
}

const CartPreviewCard: React.FunctionComponent<ICartPreviewCardProps> = (props) => {
  const item = props
  return (
    <>
      <div className='my-3 d-flex flex-row justify-content-between text-muted lh-1'>
        <p className='m-0'>
          {item.item.model} - {item.item.serviceName}
        </p>
        <p className='m-0 text-primary'>â‚¹ {item.item.price}</p>
      </div>
      <hr className='m-0' />
    </>
  )
}

export default CartPreviewCard
