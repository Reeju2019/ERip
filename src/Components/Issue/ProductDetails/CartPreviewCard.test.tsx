import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import CartPreviewCard from './CartPreviewCard'

interface Data {
    model?: string
    serviceImage: string
    serviceName: string
    price: number
    off: number
    repairTime: string
    warranty: string
    repairType: string
}
const data: Data = {
    model: "",
    serviceImage: "",
    serviceName: "",
    price: 0,
    off: 0,
    repairTime: "",
    warranty: "",
    repairType: ""
}

test('cart preview card', () => {
  render(<CartPreviewCard item={data}/>)
})
describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<CartPreviewCard item={data} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<CartPreviewCard item={data} />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
