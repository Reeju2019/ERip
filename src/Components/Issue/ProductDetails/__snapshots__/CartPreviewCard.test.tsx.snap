// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
Array [
  <div
    className="my-3 d-flex flex-row justify-content-between text-muted lh-1"
  >
    <p
      className="m-0"
    >
       - 
    </p>
    <p
      className="m-0 text-primary"
    >
      â‚¹ 
      0
    </p>
  </div>,
  <hr
    className="m-0"
  />,
]
`;
